tests/test_deserialization.py

```diff
@@ -1304,6 +1303,60 @@ class MySchema(Schema):
         errors = MySchema(partial=True).validate({'foo': 3}, partial=('bar', 'baz'))
         assert errors == {}
 
+    def test_structured_dict_nested_value_deserialization(self):
+        class BlogUserSchema(Schema):
+            email = fields.Email()
+            birthdate = fields.DateTime()
+
+        class SimpleBlogSerializer(Schema):
+            title = fields.String()
+            authors = fields.Dict(
+                keys=fields.String(),
+                values=fields.Nested(BlogUserSchema)
+            )
+
+        result = SimpleBlogSerializer().load({
+            'title': 'Gimme Shelter',
+            'authors': {
+                'Mick': {
+                    'email': 'mick@stones.com',
+                    'birthdate': '1943-07-26T00:00:00+00:00'
+                },
+                'Keith': {
+                    'email': 'keith@stones.com',
+                    'birthdate': '1943-12-18T00:00:00+00:00'
+                },
+            }
+        })
+        assert result['authors']['Mick']['email'] == 'mick@stones.com'
+        assert_datetime_equal(result['authors']['Keith']['birthdate'],
+                              dt.datetime(1943, 12, 18))
+
+        with pytest.raises(ValidationError) as excinfo:
+            SimpleBlogSerializer().load({
+                'title': 'Gimme Shelter',
+                'authors': {
+                    'Mick': {
+                        'email': 'mick-stones-com',
+                        'birthdate': 'anytime'
+                    },
+                    ('whatever', ): {
+                        'email': 'keith@stones.com',
+                        'birthdate': 'a while ago'
+                    },
+                }
+            })
+
+        assert excinfo.value.args[0] == {
+            'authors': {
+                'Mick': {'value': {
+                    'email': ['Not a valid email address.'],
+                    'birthdate': ['Not a valid datetime.']}},
+                ('whatever', ): {
+                    'key': ['Not a valid string.'],
+                    'value': {'birthdate': ['Not a valid datetime.']}}
```

Comment:
This test seems really big. Is it possible to boil the repro case down further? Any `ValidationError` that returns a dict for `messages` should cover the bug.

Reply:
You're right. This test is more a demo than a test. I just shortened it. I kept a success case and a failure. Do you think only the failure is relevant to the test?

Reply:
It looks like we only need to test the error from of custom validators. I was referring to the size of the data structure not the number of cases. Sorry for not being clear.

If this can be reproduced without using a nested schema it would be easier to follow. A schema with a single field that has a custom validator that either produces an error string or error dict should cover the bug.

I am OK with the tests as they are now since they match the reported issue. Any additional work would be icing on the cake. Thanks for working on this.